lighthouse-core/gather/gatherers/dobetterweb/domstats.js

```diff
@@ -55,7 +57,11 @@ function elementPathInDOM(element) {
   const path = [createSelectorsLabel(element)];
   let node = element;
   while (node) {
-    node = node.parentNode && node.parentNode.host ? node.parentNode.host : node.parentElement;
+    // Anchor elements have a .host property. Be sure we've found a shadow root
+    // host and not an anchor.
+    const shadowHost = node.parentNode && node.parentNode.host &&
```

Comment:
Is there a more standard way of detecting it's a shadow host? This seems fairly oblique.